implementations/rust/ockam/ockam_app/src/app/app_state.rs

```diff
@@ -137,10 +137,31 @@ impl AppState {
     /// Return true if the user is enrolled
     /// At the moment this check only verifies that there is a default project.
     /// This project should be the project that is created at the end of the enrollment procedure
-    pub async fn is_enrolled(&self) -> bool {
+    pub async fn is_enrolled(&self) -> Result<bool, string> {
         self.state().await.projects.default().is_ok()
     }
 
+    pub async fn is_enrolled(&self) -> Result<bool, String> {
```

Comment:
@etorreborre instead of returning `Result<bool>`, how do you feel about returning `Result<bool, String>` instead? we can use this to capture the exact error and either log it, or surface it in the warning menu item. i'm not particularly opinionated here.

Reply:
I am not sure what you mean here because the code about should not compile. The `crate:Result<T>` type is a type alias for `miette::Result<T, Error>` so can only specify one type for `T`. If there is an error happening in the implementation of `is_enrolled` you should be able to transform it using the `miette!` macro. See for example [this line](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_app/src/enroll/enroll_user.rs#L98).

Reply:
Ack. I'm going to stick with Result<bool>